service/controller/clusterapi/v17/resources/awsclusterconfig/create.go

```diff
@@ -36,6 +39,29 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
+	var machineDeployments []clusterv1alpha1.MachineDeployment
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding MachineDeployments for tenant cluster")
+
+		l := metav1.AddLabelToSelector(
+			&v1.LabelSelector{},
+			label.Cluster,
+			key.ClusterID(&cr),
+		)
+		o := metav1.ListOptions{
+			LabelSelector: labels.Set(l.MatchLabels).String(),
```

Comment:
@rossf7 I heard you might have similar piece of code. For some reason `String()` on `&v1.LabelSelector{}` doesn't return valid string version of label selector and you need to have `labels.Set(x).String()` indirection there. Just FYI :slightly_smiling_face:

Reply:
@tuommaki Ack & thx! Yes, this came up earlier in the loadtest e2e test I'm working on.